installer/mkpart.sh

```diff
@@ -0,0 +1,492 @@
+    freespace=$((newstatesize/(2*1024)-statefulallocated/1024))
+    echo "        Leftover free space: $freespace MB"
+    echo "    crouton: $((croutonsize/(2*1024))) MB"
+    echo "----"
+
+    echo -n "Type 'yes' if you are you satisfied with these new sizes: "
+    if [ -t 0 ]; then
+        read -r line
+    else
+        line="$CROUTON_MKPART_YES"
+        echo "$line"
+    fi
+    if [ "$line" != "yes" ]; then
+        error 2 "Aborting..."
+    fi
+fi
+
+if grep -q '^[^ ]* '"`readlink -m '/var/run/crouton'`/" /proc/mounts \
+        || grep -q '^[^ ]* '"$PREFIX"'/chroots' /proc/mounts; then
+    error 1 \
+"Some chroots are mounted. Log out from these, and run this script again."
+fi
+
+echo "====== WARNING ======"
+if [ -n "$DELETE" ]; then
+    echo "This script will now log you off and wipe '$name' (${ROOTDEVICEPREFIX}$CROUTONPARTNUMBER)."
+elif [ -z "$NOCREATE" ]; then
+    echo "This script will now log you off, setup the crouton partition, and reboot."
+else
+    echo "This script will now log you off and unmount the stateful partition."
+fi
+echo "Make sure all your current work is saved."
+
+time=10
+while [ "$time" -gt 0 ]; do
+    printf "\\rYou have %2d seconds to press Ctrl-C to abort..." "$time"
+    sleep 1
+    time="$((time-1))"
+done
+echo
+
+# Unmount crouton partition, if it exists
+if [ -n "$DELETE" -o -n "$NOCREATE" ]; then
+    for mountpoint in `awk \
+            '$1 == "'"${ROOTDEVICEPREFIX}$CROUTONPARTNUMBER"'" { print $2 }' \
+            /proc/mounts`; do
+        if ! umount "$mountpoint"; then
+            error 1 \
+"Cannot unmount partition in '$mountpoint', make sure nothing is using it."
+        fi
+    done
+fi
+
+( # Fork a subshell, with input/output in VT $LOGVT
+    clear || true
+
+    # Redefine error function
+    error() {
+        local ecode="$1"
+        shift
+        chvt $LOGVT
+        echo "$*" 1>&2
+        echo "Press enter to reboot..."
+        sync
+        ( sleep 30; reboot ) &
+        read -r line
+        settrap ""
+        reboot
+        exit "$ecode" # unreachable
+    }
+
+    addtrap "chvt $LOGVT; echo 'Something went wrong, press enter to reboot.'; \
+             sync; ( sleep 30; reboot ) & read -r line; \
+             echo 'Rebooting...'; reboot"
+
+    # Make sure screen does not blank out (setterm does not work in this
+    # context: send control sequence instead)
+    /bin/echo -n -e "\x1b[9;0]\x1b[14;0]"
+
+    # Get out of the stateful partition
+    cd /
+
+    echo "Logging user out..."
+    dbus-send --system --dest=org.chromium.SessionManager --type=method_call \
+        --print-reply /org/chromium/SessionManager \
+        org.chromium.SessionManagerInterface.StopSession \
+        string:"crouton installer"
+
+    # Detect when the user has been logged out (1 minute timeout)
+    tries=12
+    while [ "$tries" -gt 0 ] && grep -q '^/home/.shadow' /proc/mounts; do
+        chvt $LOGVT
```

Comment:
@DennisLfromGA @dnschneid @drinkcat 

Ok so here's where it fails. The trap set up by ```addtrap``` above caused the reboot.

I tried to manually run ```chvt 9```. It returned an error saying ```Couldn't get a file descriptor referring to the console```.

Seems it is a upstream change( Maybe? Just guessing... ) Because my old pixel which has the separate partition set up already cannot run ```chvt``` in command line either.
I haven't check whether I can run the ```separate_partition``` branch on my old machine with the new system again now but I guess it would fail.